### ISSUE TITLE
Checklist for Model Hub

### ISSUE THREAD

**Issue by prajjwal1** [huggingface/transformers#6762]
# 🚀 Feature request

I want to propose a checklist for model hub. Model hub is increasingly being used by `transformers` users. It is imperative that users exactly know what model they are downloading and importing for transparency and research-related reasons. It should be mandatory for users to upload a `REAME.md` while uploading model on model hub. Additionally, it will reduce the need to send a seperate PR for model cards.

Checklist should contain the following elements:
- Details of training and validation set used. Specify the split if not using default.
- A table with results on the validation set
- Model name (can be inferred from `config.json`)
- Objective used for training
- Hyperparameter configurations if default ones are not being used
- Link to implemented code (if available, and is different from default examples)
- For how many epochs the model was trained for ?
- Associated paper (if published)

These are just the general ones. Happy to hear about other suggestions. This is just a suggestion which would help many users. Thanks for this feature.

**Comment by julien-c** [comment]
Yes! This is something we'll build, step by step. I'll post more about our roadmap for this in the coming weeks.

**Comment by BramVanroy** [comment]
> Yes! This is something we'll build, step by step. I'll post more about our roadmap for this in the coming weeks.

Maybe you can provide a web-based interface with input fields/drop downs that can generate the model card automatically.

**Comment by prajjwal1** [comment]
posting so that the bot doesn't close this issue.

**Comment by julien-c** [comment]
@prajjwal1 Have you checked out our new model storage/model versioning features on huggingface.co? You can also now edit your model card directly from the website (or from `git`) and we'll make this workflow more prominent (vs. adding model cards to `transformers`) in the coming days/weeks.

Feedback welcome, here or on the Forum.

**Comment by prajjwal1** [comment]
No, I haven't. I am aware of it. I will check out soon.
